common/configParser_test.go

```diff
@@ -46,46 +316,179 @@ func TestLoadGasScheduleConfig(t *testing.T) {
-	file, err := os.Create("testGasSchedule.toml")
-	assert.Nil(t, err)
+		filePath := path.Join(t.TempDir(), "testGasSchedule.toml")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
 
-	_, _ = file.WriteString(testString)
-	_ = file.Close()
+		_, err = file.WriteString(testString)
+		assert.Nil(t, err)
 
-	gasSchedule, err := common.LoadGasScheduleConfig("testGasSchedule.toml")
-	assert.Nil(t, err)
+		gasSchedule, err := common.LoadGasScheduleConfig(filePath)
+		assert.Nil(t, err)
 
-	if _, err = os.Stat("testGasSchedule.toml"); err == nil {
-		_ = os.Remove("testGasSchedule.toml")
-	}
+		expectedGasSchedule := make(map[string]uint64)
+		expectedGasSchedule["GetFunction"] = 84
+		expectedGasSchedule["StorageStore"] = 11
+		expectedGasSchedule["BigIntNew"] = 9
+		expectedGasSchedule["BigIntAdd"] = 44
+		expectedGasSchedule["Drop"] = 1
+		expectedGasSchedule["I64Load"] = 12
+		expectedGasSchedule["MemoryCopy"] = 14623
 
-	expectedGasSchedule := make(map[string]uint64)
-	expectedGasSchedule["GetFunction"] = 84
-	expectedGasSchedule["StorageStore"] = 11
-	expectedGasSchedule["BigIntNew"] = 9
-	expectedGasSchedule["BigIntAdd"] = 44
-	expectedGasSchedule["Drop"] = 1
-	expectedGasSchedule["I64Load"] = 12
-	expectedGasSchedule["MemoryCopy"] = 14623
-
-	assert.Equal(t, expectedGasSchedule, gasSchedule["GasSchedule"])
+		assert.Equal(t, expectedGasSchedule, gasSchedule["GasSchedule"])
+	})
 }
 
 func TestLoadRoundConfig(t *testing.T) {
-	testString := ""
+	t.Parallel()
+
+	t.Run("invalid file should error", func(t *testing.T) {
+		t.Parallel()
 
-	file, err := os.Create("testEnableRounds.toml")
-	assert.Nil(t, err)
+		conf, err := common.LoadRoundConfig("testFile01")
+		assert.Nil(t, conf)
+		assert.Error(t, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
 
-	_, _ = file.WriteString(testString)
-	_ = file.Close()
+		testString := ""
 
-	roundActivationConfig, err := common.LoadRoundConfig("testEnableRounds.toml")
-	assert.Nil(t, err)
+		filePath := path.Join(t.TempDir(), "testEnableRounds.toml")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString(testString)
+		assert.Nil(t, err)
+
+		roundActivationConfig, err := common.LoadRoundConfig(filePath)
+		assert.Nil(t, err)
+
+		if _, err = os.Stat("testEnableRounds.toml"); err == nil {
+			_ = os.Remove("testEnableRounds.toml")
+		}
+
+		require.Equal(t, &config.RoundConfig{}, roundActivationConfig)
+	})
+}
+
+func TestLoadEpochConfig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("invalid file should error", func(t *testing.T) {
+		t.Parallel()
+
+		conf, err := common.LoadEpochConfig("testFile01")
+		assert.Nil(t, conf)
+		assert.Error(t, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		testString := `
+[EnableEpochs]
+    SCDeployEnableEpoch = 1
+`
+
+		filePath := path.Join(t.TempDir(), "testFile02")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString(testString)
+		assert.Nil(t, err)
```

Comment:
missing call to Close? Valid on all occurrences.

Reply:
added